unitex/src/fr/umlv/unitex/tfst/tagging/TaggingModel.java

```diff
@@ -105,27 +150,677 @@ protected void updateModel() {
+				
+				BufferedReader br = new BufferedReader( new InputStreamReader( new FileInputStream(alphabetFile) , "UTF16"));
+				String line; 
+				while ((line = br.readLine()) != null && !line.equals("\r\n")) {
+					if( line.startsWith("#")) {
+						alphabet.append(line.charAt(0)).append("-").append(line.charAt(1));
+					}		
+					alphabet.append(line.trim());
+				}
+				StringBuilder regexBuilder = new StringBuilder();
+				regexBuilder.append("(<E>)|([").append(alphabet.toString())
+					.append("]+|[^").append(alphabet.toString()).append("]+)");
+				regex = regexBuilder.toString();
+			} catch (IOException e) {
+				 //TODO Auto-generated catch block
+				System.out.println("exception");
+			} finally {}	
+		}	
+	}
+	
+	/*
+	 * for each box we will start at it's starting bound, split the inside based on delimiters and add them to the hashmap. 
+	 * if 2 words split by a comma are found, they will be stored subsequently in the hashmap at the starting bound's value, then +1 and so on. 
+	 */
+	public void generateTokensList() {
+		for( TfstGraphBox gb : boxes ) {
+			if( taggingStates[gb.getBoxNumber()] == TaggingState.NOT_PREFERRED || taggingStates[gb.getBoxNumber()] == TaggingState.USELESS )
+				break;
+			if( gb == null || gb.getContent() == null )
+				continue;
+			
+			if( gb.getBounds() == null)
+				continue;
+			
+			int index = gb.getBounds().getStart_in_tokens();
+			String temp = null;
+			ArrayList<String> tokensList = new ArrayList<String>();
+			if( gb.getContent().contains("{")  )
+				temp = gb.getContent().split(",")[0].substring(1);
+			else {
+				temp = gb.getContent();
+			}
+			/* ERROR WITH KOREAN MODE
+			 * Pattern pattern = Pattern.compile(regex);
+			Matcher matcher = pattern.matcher(temp);
+			
+			while( matcher.find() ) {
+				tokensList.add(matcher.group());
+			}*/
+			
+                        if( gb.getBounds().getEnd_in_tokens()
+			== gb.getBounds().getStart_in_tokens() + tokensList.size() - 1) {
+			
+		    for( String s : tokensList ) {
+	            	if( !tokens.containsKey(index) )  
+	            		tokens.put(index, new ArrayList<String>());
+	            	if( !tokens.get(index).contains(s))
+	            		tokens.get(index).add(s);
+	            	index++;
+	            }
+			} else
+			if( gb.getBounds().getEnd_in_tokens()
+			== gb.getBounds().getStart_in_tokens() ){
+				if( !tokens.containsKey(index) )  
+            		tokens.put(index, new ArrayList<String>());
+            	if( !tokens.get(index).contains(temp))
+            		tokens.get(index).add(temp);
+			} else {
+				// implicit case where the difference is 0 < D < tokensList.length
+				tokens.put(index, new ArrayList<String>());
+				tokens.get(index).add(temp);
+			}
+			
+		}
+		for ( int i=0; i<tokens.size();i++ )
+	    	if( !tokens.containsKey(i) ) {
+	    		tokens.put( i, new ArrayList<String>());
+	    		tokens.get(i).add(" ");
+	    }
+	}
+	
+	/**  This function creates all the paths we can have between the first box
```

Comment:
add what this method does, I think it is, marking as TO_CHECK the USELESS boxes between bfp and bfs that have become accessible and co-accessible